mod/wikirate_source/set/type/source/source_preview.rb

```diff
@@ -148,142 +132,22 @@ def valid_mime_type? mime_type
-      <<-HTML
-        <a href="#{file_card.attachment.url}" class="btn btn-primary" role="button">Download</a>
-      HTML
-    else
-      url_card = card.fetch(trait: :wikirate_link)
-      url = url_card ? url_card.item_names.first : nil
-      <<-HTML
-        <a href="#{preview_url}" class="btn btn-primary" role="button">Visit Original Source</a>
-      HTML
+    wrap_with :div, class: "hidden" do
+      [
+        wrap_with(:div, card.cardname.url_key, id: "source-name"),
+        wrap_with(:div, preview_url, id: "source_url"),
+        wrap_with(:div, args[:year], id: "source-year"),
+        wrap_with(:div, args[:company], id: "source_company"),
+        wrap_with(:div, args[:topic], id: "source_topic")
+      ]
     end
   end
 
-  def source_details_html
-    <<-HTML
-      <li role="presentation" class="active" >
-        <a class='' data-target="#tab_details" data-toggle="source_preview_tab_ajax">
-          <i class="fa fa-info-circle"></i> <span>Source Details</span>
-        </a>
-      </li>
-    HTML
-  end
-
-  def claim_tab_html
-    <<-HTML
-      <li role="presentation" >
-        <a class='' data-target="#tab_claims" data-toggle="source_preview_tab_ajax"  href='/#{card.cardname.url_key}+source_note_list?slot[hide]=header,menu' >
-            <i class='fa fa-quote-left'><span id="claim-count-number " class="count-number">#{note_count}</span></i><span>#{Card[ClaimID].name.pluralize}</span>
-        </a>
-      </li>
-    HTML
-  end
-
-  def metric_tab_html
-    <<-HTML
-       <li role="presentation" >
-        <a class='' data-target="#tab_metrics" data-toggle="source_preview_tab_ajax" href='/#{card.cardname.url_key}+metric_search?slot[hide]=header,menu' >
-          <i class="fa fa-bar-chart">
-          <span id="metric-count-number" class="count-number">
-            #{metric_count}
-          </span>
-          </i>
-          <span>#{Card[MetricID].name.pluralize}</span>
-        </a>
-      </li>
-    HTML
-  end
-
-  def link_button url
-    <<-HTML
-      <li role="presentation" >
-        <a class='' href='#{url}' target="_blank">
-          <i class="fa fa-external-link-square"></i> Visit Original
-        </a>
-      </li>
-    HTML
-  end
-
-  def file_download_button
+  view :non_previewable, tags: :unknown_ok do |_args|
     file_card = card.fetch trait: :file
-    <<-HTML
-      <li role="presentation" >
-        <a class='' href='#{file_card.attachment.url}' download>
-          <i class="fa fa-download" aria-hidden="true"></i> Download
-        </a>
-      </li>
-    HTML
-  end
-
-  view :preview_options, tags: :unknown_ok do
-    url = preview_url
-    result = source_details_html
-    result += claim_tab_html
-    result += metric_tab_html
-    result +=
-      case card.source_type_codename
-      when :wikirate_link
-        link_button url
-      when :file
-        file_download_button
-      else
-        ""
-      end
-    result
+    url, text = if file_card [file_card.attachment.url, "Download"]
```

Comment:
Is this valid ruby? Condition and statement in the same line without any separation like line break or `then`